app/controllers/concerns/idv/document_capture_concern.rb

```diff
@@ -50,6 +50,35 @@ def selfie_requirement_met?
         stored_result.selfie_check_performed?
     end
 
+    # @returns[String] The document capture path for the current vendor
+    def doc_auth_vendor_capture_path(in_hybrid_mobile)
+      case doc_auth_vendor
+      when Idp::Constants::Vendors::SOCURE
+        in_hybrid_mobile ? idv_hybrid_mobile_socure_document_capture_path
+                         : idv_socure_document_capture_path
+      when Idp::Constants::Vendors::LEXIS_NEXIS, Idp::Constants::Vendors::MOCK
+        in_hybrid_mobile ? idv_hybrid_mobile_document_capture_path
+                         : idv_document_capture_path
+      end
+    end
+
+    def ensure_user_stays_in_vendor_bucket
+      in_hybrid_mobile = request.fullpath.include?('/hybrid_mobile/')
+      expected_path = doc_auth_vendor_capture_path(in_hybrid_mobile)
+
+      if I18n.locale == :en
+        actual_path = request.path
+      else
+        actual_path = "/#{I18n.locale}#{request.path}"
+      end
+
+      puts "ensure_user_stays_in_vendor_bucket: expected_path: #{expected_path}"
+      puts "                                      actual_path: #{actual_path}"
+      puts "                                 request.fullpath: #{request.fullpath}"
+
+      redirect_to expected_path if actual_path != expected_path
```

Comment:
This seems like it could be kind of brittle, can we do a comparison on the expected `doc_auth_vendor` and the calculated doc auth vendor instead closer to the controller actions and use the context there to pass in the relevant variables?

```ruby
before_action ->{ redirect_to_correct_vendor(Idp::Constants::Vendors::SOCURE, true) }, only: [:relevant_action]
before_action ->{ redirect_to_correct_vendor(Idp::Constants::Vendors::SOCURE, false) }, only: [:other_relevant_action]
before_action ->{ redirect_to_correct_vendor(Idp::Constants::Vendors::LEXIS_NEXIS, true) }, only: [:relevant_action]
before_action ->{ redirect_to_correct_vendor(Idp::Constants::Vendors::LEXIS_NEXIS, false) }, only: [:other_relevant_action]

# concern.rb
def redirect_to_correct_vendor(vendor, in_hybrid_mobile)
      expected_doc_auth_vendor = doc_auth_vendor
      return if vendor == expected_doc_auth_vendor
      path = case expected_doc_auth_vendor
      when Idp::Constants::Vendors::SOCURE
        in_hybrid_mobile ? idv_hybrid_mobile_socure_document_capture_path
                         : idv_socure_document_capture_path
      when Idp::Constants::Vendors::LEXIS_NEXIS, Idp::Constants::Vendors::MOCK
        in_hybrid_mobile ? idv_hybrid_mobile_document_capture_path
                         : idv_document_capture_path
      end
   redirect_to path
```

Reply:
Better. Thanks.

Reply:
Leaving the actions clauses out, because based on conversation with @amirbey when refining this, he really does mean it to check all the actions.

Reply:
In reading through the controller, there's only one display action it seems? Is there value in redirecting the `update` request that should never happen?

One of the [controllers](https://github.com/18F/identity-idp/pull/11425/files#diff-584bb3ff7c4a2db11831e735f0a1a31486a0a20d0572b5ddd01b8ddb1fd960d8L22-L27) already has the following, which might conflict?

```ruby
      case doc_auth_vendor
      when Idp::Constants::Vendors::SOCURE
        redirect_to idv_socure_document_capture_url
      when Idp::Constants::Vendors::LEXIS_NEXIS, Idp::Constants::Vendors::MOCK
        render :show, locals: extra_view_variables
      end
```

Reply:
_All_ of these are "should never happen" conditions; I don't see that any one has priority over the others.

Reply:
That code was the result of addressing similar concerns during the initial implementation. Removed, as it is now redundant.

Reply:
> _All_ of these are "should never happen" conditions; I don't see that any one has priority over the others.

Ah, with the understanding that this is more about handling circumvention, is there a way to add these checks to `step_info` so that `confirm_step_allowed` can handle it automatically?

Reply:
(apologies, unresolved this thread following the change in my understanding)

Reply:
Hmm. Will look into. Implemented your suggestion above; it took a little tweaking to allow for the mock proofer.
But putting it in `confirm_step_allowed` is an interesting thought.

Reply:
Hit the timebox for doing this with the step_info and confirm_step_allowed approach. It's a good idea, and we have a ticket for it: LG-14990